core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -87,7 +91,18 @@
                String operation,
                Map<String, String> summary,
                List<ManifestFile> dataManifests) {
-    this(io, INITIAL_SEQUENCE_NUMBER, snapshotId, parentId, timestampMillis, operation, summary, null);
+    this(io, snapshotId, parentId, timestampMillis, operation, summary, null, dataManifests);
+  }
+
+  BaseSnapshot(FileIO io,
```

Comment:
Do we need to introduce an additional constructor? Since these constructors are package-private, we can remove them if we can refactor all of the uses.

Reply:
This was actually me trying to avoid changing a lot of existing code since there are a lot of usage for the old constructor; but since they are all in tests and I'd expect less merge conflicts from tests, I think it might be fine to directly change. Will update.